### DISCUSSION TITLE
members_dict = namelist_config[listname]['members']KeyError in namelistdescription.py

### DISCUSSION THREAD

**Discussion by jonnyhtw** [MetOffice/simulation-systems#139] [LFRic]
hi there, 

i'm porting an LFRic GC5 climate workflow to our HPC in new zealand and am getting the following error in several build tasks...

![image](https://github.com/MetOffice/simulation-systems/assets/12940715/e203e102-d3df-4786-9ca4-78677d896f5e)

essentially, the code on the left fails with...

`members_dict = namelist_config[listname]['members']KeyError`

... whereas the code on the right works and is adapted from from a previous version of the code which worked fine.

the full `fcm` path of the failing file is...

`fcm:lfric.xm/trunk/infrastructure/build/tools/configurator/namelistdescription.py@46736`

and the previous version of the code which worked is...

`fcm:lfric.xm/trunk/infrastructure/build/tools/configurator/namelistdescription.py@42090`

the relevant lines from the output of `fcm di fcm:lfric.xm/trunk/infrastructure/build/tools/configurator/namelistdescription.py -r :42090:46736` are...

<img width="860" alt="image" src="https://github.com/MetOffice/simulation-systems/assets/12940715/0692208c-dd13-4d29-9ba6-601158bf12cd">

thanks for any ideas!

all the best, 

jonny

**Comment by stevemullerworth** [comment]
Hi Jonny,

Based on your version numbers, I suspect you need version 2.0.0 of rose_picker which was released about 5 months ago. It can be found in the gpl_utilities project within the LFRic repository. 

`svn export --username <SRS username> https://code.metoffice.gov.uk/svn/lfric/GPL-utilities/trunk $INSTALL_DIR/opt/rose-picker`

**Reply by jonnyhtw** [reply]
hi @stevemullerworth, 

brilliant, using `Rose_Picker 2.0.0` worked perfectly!

thanks a lot,

jonny
